### PR TITLE
add kubernetes_job_timeout parameter

### PR DESCRIPTION
**Pull request by VMois**
closes reanahub/reana-job-controller#343

How to test:

This PR is part of a bigger group of PRs. Please refer to the addressed issue to see all of them.

1. Modify Serial helloworld example, with `kubernetes_job_timeout=20` (in seconds) and `sleeptime=2` (2 seconds so the workflow will definitely run longer than 20 seconds):

`reana.yml`
```yaml
...
parameters:
    helloworld: code/helloworld.py
    inputfile: data/names.txt
    outputfile: results/greetings.txt
    sleeptime: 2  # <= here 
...
steps:
      - environment: 'python:2.7-slim'
        kubernetes_job_timeout: 20  # <= here
...
```

2. Start workflow `reana-client run -w serial-timeout -f reana.yaml`

3. After 25-30 seconds, check using `reana-client list` if workflow failed. It should.

4. Using `reana-client logs` check if the reason is `Job was killed due to timeout.`

5. Try to input string or object instead of integer for `kubernetes_job_timeout`, the workflow should fail with an error message like:

```
('Error while marshalling value=not_correct_timeout to type=integer/int32.', ValueError("invalid literal for int() with base 10: 'not_correct_timeout'"))
```

Check it using `reana-client logs`.

**Note:** You will see an error message above, but you will not see engine logs in the `reana-client logs` output. This is due to [this issue](https://github.com/reanahub/reana-workflow-controller/issues/426). It will be fixed after this PR.

### CODE REVIEW

File: reana_workflow_engine_serial/tasks.py [hunk 1]
```diff
@@ -117,6 +117,7 @@ def run_step(
             unpacked_image=step.get("unpacked_image", False),
             kubernetes_uid=step.get("kubernetes_uid", None),
             kubernetes_memory_limit=step.get("kubernetes_memory_limit", None),
+            kubernetes_job_timeout=step.get("kubernetes_job_timeout", None),
```
Comment (mvidalgarcia) [inline reana_workflow_engine_serial/tasks.py:120]: `None` is the default backup value, it shouldn't be needed. The same applies to other options.

**Review by mvidalgarcia — COMMENTED** [review]
LGTM in general.

```console
$ reana-client logs -w serial-timeout  

==> Job logs
==> Step: 4ed87f77-3a97-4589-85c8-adb50311e54b
==> Workflow ID: ff997494-b681-4f51-8103-0136e749e0fe
==> Compute backend: Kubernetes
==> Job ID: reana-run-job-921ca749-2ec2-4326-8f61-076612aba566
==> Docker image: python:2.7-slim
==> Command: sleep 25 && python "code/helloworld.py" --inputfile "data/names.txt" --outputfile "results/greetings.txt" --sleeptime 0
==> Status: failed
==> Started: 2021-12-16T16:34:55
==> Finished: 2021-12-16T16:35:25
==> Logs:
job: :
 2021-12-16T16:35:21 INFO     Parameters: inputfile=data/names.txt outputfile=results/greetings.txt sleeptime=0.0


Completed

Job was killed due to timeout.
```

However, I've found a really strange behavior when fetching logs from old runs.
If I run a failing `serial-timeout` workflow multiple times, the logs from the latest run (`reana-cilent logs -w serial-timeout`) look as expected ("Job was killed due to timeout."). But if I check an old run, the logs are gone (?). E.g.

```console
$ reana-client logs -w serial-timeout.2


==> Job logs
==> Step: d7d272e7-75c5-49e3-a430-91d5f3442c2e
==> Workflow ID: 7be8afc1-1485-4a50-a1e1-37edc7207698
==> Compute backend: Kubernetes
==> Job ID: reana-run-job-f5f517fe-340e-4dc7-877c-b36b0b3a9975
==> Docker image: python:2.7-slim
==> Command: sleep 25 && python "code/helloworld.py" --inputfile "data/names.txt" --outputfile "results/greetings.txt" --sleeptime 0
==> Status: failed
==> Started: 2021-12-16T16:19:42
==> Finished: 2021-12-16T16:20:06
==> Step d7d272e7-75c5-49e3-a430-91d5f3442c2e emitted no logs.
```

Perhaps it's not even related to your changes. Can you reproduce it?

**Edit:** I've just found out that the logs are gone whenever one deletes de hanging k8s failed jobs, so this is clearly an issue.

```console
$ reana-client run -w yadage-timeout -f reana-yadage.yaml
... # wait until it terminates
$ reana-client list
NAME             RUN_NUMBER   CREATED               STARTED               ENDED                 STATUS  
yadage-timeout   1            2021-12-17T11:41:35   2021-12-17T11:41:46   2021-12-17T11:42:55   failed  
$ reana-client logs -w yadage-timeout
...
Completed

Job was killed due to timeout.
$ kubectl delete jobs.batch --all  # clean-up failed jobs
$ reana-client logs -w yadage-timeout
...
==> Step fitdata emitted no logs.
```

File: reana_workflow_engine_serial/tasks.py (hunk 1 above)
Comment (VMois) [reply reana_workflow_engine_serial/tasks.py]: Good catch. Done

**Review by mvidalgarcia — APPROVED** [review]
Tested multiple scenarios, works as expected.

**Review by tiborsimko — APPROVED** [review]

**Pull request merged**